### ISSUE TITLE
Support for mbedtls

### ISSUE THREAD

**Issue by lgrahl** [rawrtc/rawrtc#121]
This would drop the gigantic OpenSSL dependency requirement. Requires work in re.

**Comment by lgrahl** [comment]
Low priority though. The biggest part of RAWRTC when building statically is usrsctp, so that should be our focus.

**Comment by fire** [comment]
See https://github.com/sctplab/usrsctp/pull/579

**Comment by lgrahl** [comment]
Thanks but the bigger issue was and still is re :)

**Comment by dhuawong** [comment]
Hi, 
Is there any new progress in mbedtls support? We are running rawrtc on embedded systems and openssl is too bulky for us.

**Comment by fire** [comment]
We looked into https://github.com/paullouisageneau/libdatachannel and their source was relicensed to MPL2.0. libdatachannel switched to mbedtls.

**Comment by lgrahl** [comment]
> Hi, Is there any new progress in mbedtls support? We are running rawrtc on embedded systems and openssl is too bulky for us.

No progress in re regarding support for mbedtls that I know of.

I'm happy to maintain rawrtc but since I currently have no use for it myself, contributions will need to be made to advance on issues.

Rewriting the use of the underlying event loop to e.g. libuv (or better, an abstraction to use an arbitrary event loop) would most likely unblock a lot of issues, including this one.
